src/module.c

```diff
@@ -4848,9 +4853,18 @@ void moduleHandleBlockedClients(void) {
             ctx.module = bc->module;
             ctx.client = bc->client;
             ctx.blocked_client = bc;
+            const ustime_t start = server.ustime;
             bc->reply_callback(&ctx,(void**)c->argv,c->argc);
+            reply_us = ustime()-start;
             moduleFreeContext(&ctx);
         }
+        /* Update stats now that we've finished the blocking operation.
+         * This needs to be out of the reply callback above given that a
+         * module might not define any callback and still do blocking ops.
+         */
+        if (c && !bc->blocked_on_keys) {
+            updateStatsOnUnblock(c, c->background_duration, reply_us);
+        }
```

Comment:
i seem to be missing the place that calls updateStatsOnUnblock for modules that **are** blocked on keys, so i'll comment here although not related to this lines.

maybe the solution is that instead adding a call in the individual serveClientsBlockedOnListKey serveClientsBlockedOnSortedSetKey, and serveClientsBlockedOnStreamKey, you'll do that once in handleClientsBlockedOnKeys, this way it'll also handle modules blocked on keys.

Reply:
@oranagra I believe we can't use `handleClientsBlockedOnKeys()` given that we want to update the per receiver stats ( `client *receiver()` ) and we only iterate over the receivers within each of those methods correct? wdyt?

Reply:
PS: added the reply time tracking to `moduleHandleBlockedClients`

Reply:
yeah, you're right.